src/algo/depth-to-rgb-calibration/valid-scene.cpp

```diff
@@ -679,5 +679,137 @@ bool optimizer::is_scene_valid()
                                               &_z.dir_ratio1 );
 
     //return((!res_movement) && res_edges && res_gradient);
-    return(!res_movement);
+
+    auto valid = !res_movement;
+    if (!_manual_trigger)
+        valid = valid && input_validity_checks(data);
+        
+    return(valid);
+}
+
+bool optimizer::input_validity_checks(input_validity_data* data )
+{
+    auto dir_spread = check_edges_dir_spread(_z);
+    auto saturation = check_for_saturation(_ir);
```

Comment:
This is confusing.

'saturation' suggests that a 'true' value means the image is saturated, but it seems the return value from this function returns 'true' for a valid scene, meaning UNsaturated.